cmd/bootkube/render.go

```diff
@@ -100,7 +100,8 @@ func validateRenderOpts(cmd *cobra.Command, args []string) error {
 		return errors.New("You must specify either all or none of --etcd-ca-path, --etcd-certificate-path, and --etcd-private-key-path")
 	}
 	if renderOpts.etcdCertificatePath != "" && renderOpts.selfHostedEtcd {
-		return errors.New("Cannot specify --etcd-certificate-path with --experimental-self-hosted-etcd")
+		return errors.New("Cannot specify --etcd-certificate-path with --experimental-self-hosted-etcd." +
+			" Self-hosted etcd + TLS will auto-generate certs based on root CA cert.")
```

Comment:
Sorry, one last thing: you need to update the section line 176:
- Pull out the `else` branch to just always run.
- Remove the `http://` override (lines 177-180)
- Remove `defaultSelfHostedEtcdServers` and refactor that check+warning to just use the CIDR and not the full URL w/ scheme (so it's compatible with both `http` and `https` -- what it is really checking is the CIDR)